core/Pos/Aeson/Core.hs

```diff
@@ -39,14 +49,56 @@ deriving instance ToJSON SlotCount
 instance FromJSON CoinPortion where
     parseJSON v = unsafeCoinPortionFromDouble <$> parseJSON v
 
+instance ToJSON CoinPortion where
```

Comment:
I have yet to arrive at the bottom of the PR, so I beg your pardon if this is contemplated later, but essentially I would like to see, every time we add _separated_ `ToJSON` and `FromJSON` instances (i.e. not generated atomically like in `deriveJSON`), roundtrip properties (as in `decode $ encode myCoinPortion === Just myCoinPortion`). As we are fiddling with doubles (it seems) I think it's important to have confidence we are not losing precision. Similar considerations can be applied to `Timestamp`.